opticalFlow/lucasKanadeOpticalFlow.py

```diff
@@ -25,106 +24,160 @@
 previewWindow = True
 
 # output video params
-fps = 20 # fps of output video, should match input video
+fps = 20  # fps of output video, should match input video
 
 # visualization parameters
-numPts = 5 # max number of points to track
-trailLength = 60 # how many frames to keep a fading trail behind a tracked point to show motion
-trailThickness = 8 # thickness of the trail to draw behind the target
-trailFade = 4 # the intensity at which the trail fades
-pointSize = 15 # pixel radius of the circle to draw over tracked points
+numPts = 5  # max number of points to track
+trailLength = 60  # how many frames to keep a fading trail behind a tracked point to show motion
+trailThickness = 8  # thickness of the trail to draw behind the target
+trailFade = 4  # the intensity at which the trail fades
+pointSize = 15  # pixel radius of the circle to draw over tracked points
 
 # params for Shi-Tomasi corner detection
 shitomasi_params = {
     "qualityLevel": 0.3,
     "minDistance": 7,
-    "blockSize": 7
+    "blockSize": 7,
 }
 
 # params for Lucas-Kanade optical flow
 LK_params = {
-    "winSize": (15,15),
+    "winSize": (15, 15),
     "maxLevel": 2,
-    "criteria": (cv.TERM_CRITERIA_EPS | cv.TERM_CRITERIA_COUNT, 10, 0.03)
+    "criteria": (
+        cv.TERM_CRITERIA_EPS
+        | cv.TERM_CRITERIA_COUNT,
+        10,
+        0.03,
+    ),
 }
 
 
 # SETUP -----------------------------------------------------------------------
 
 # generate random colors
-color = np.random.randint(0,255,(100,3))
+color = np.random.randint(0, 255, (100, 3))
 
 # read the video file into memory
 cap = cv.VideoCapture(vidpath)
 
 # get the first frame
 _, old_frame = cap.read()
-old_gray = cv.cvtColor(old_frame, cv.COLOR_BGR2GRAY)
+old_gray = cv.cvtColor(
+    old_frame, cv.COLOR_BGR2GRAY
+)
 
 # get resolution of video
 res_x = len(old_frame[0])
 res_y = len(old_frame)
 
 # create crosshair mask
-crosshair_bottom = int(0.7*res_y)
-crosshair_top = int(0.3*res_y)
-crosshair_left = int(0.3*res_x)
-crosshair_right = int(0.7*res_x)
-crosshairmask = np.zeros(old_frame.shape[:2], dtype="uint8")
-cv.rectangle(crosshairmask, (crosshair_left, crosshair_top), (crosshair_right, crosshair_bottom), 255, -1)
+crosshair_bottom = int(0.7 * res_y)
+crosshair_top = int(0.3 * res_y)
+crosshair_left = int(0.3 * res_x)
+crosshair_right = int(0.7 * res_x)
+crosshairmask = np.zeros(
+    old_frame.shape[:2], dtype="uint8"
+)
+cv.rectangle(
+    crosshairmask,
+    (crosshair_left, crosshair_top),
+    (crosshair_right, crosshair_bottom),
+    255,
+    -1,
+)
 
 # create masks for drawing purposes
-trail_history = [[[(0,0), (0,0)] for i in range(trailLength)] for i in range(numPts)]

```

Comment:
the point of a list comprehension is to be concise and compact, so not really a fan of spreading it out to multiple lines